[CUT-132] Popup to create announcement channels

sasangachathumal opened the pull request:
**Story/Ticket Reference ID from JIRA:**
https://intersective.atlassian.net/browse/CUT-132

**Additional Comments:**
`Add here`

**Checklist:**

- [x] Unit test completed?: (Y/N)
- [x] Docs folder up to date?: (Y/N)
- [x] Add if anything missed: (Y/N)

All checklist are okay.

This PR looks great - it's ready to merge! Please review and let's ship it. :)

sasangachathumal committed:
[CUT-132]
create component for announcement channel create.
create new methods for open and manage new popup.
added logic to create channels in popup components.
update demo service.

sasangachathumal committed:
[CUT-132]
updated way passing created channels to chat list component.
update colors in UI.
update chat list to show messages if chat all ready created.
update chat list to handel padding data from popup.

sasangachathumal committed:
[CUT-132]
move popup styles to main scss file.
create new component for group channel create.
added styles for new component.
added methods for new component.

shawnm0705 commented on src/app/chat/announcement-chat-popup/announcement-chat-popup.component.ts:
```diff
@@ -0,0 +1,96 @@
+    this.createdChatChannels = {};
+    this.isExpertsAnnouncementsChecked = false;
+    this.isLearnersAnnouncementsChecked = false;
+    this.timelineUuid = this.storage.getUser().timelineUuid;
```
we don't need to create a variable `timelineUuid` and assign value to it here.
Instead, we can just create a get function `get timelineUuid()` and use it wherever we need.

shawnm0705 commented on src/app/chat/announcement-chat-popup/announcement-chat-popup.component.ts:
```diff
@@ -0,0 +1,96 @@
+    this.creating = false;
+    this.createdChatChannels = {};
+    this.isExpertsAnnouncementsChecked = false;
+    this.isLearnersAnnouncementsChecked = false;
```
We need to check if those channels are created already, if so, don't let admin re-create them

sasangachathumal committed:
[CUT-132]
add changes mention in code review.
added logic to check created channels before open popups.
added logic to popups check created channels and disable items

shawnm0705 commented on src/app/chat/announcement-chat-popup/announcement-chat-popup.component.ts:
```diff
@@ -0,0 +1,114 @@
+    this.isLearnersAnnouncementsChecked = false;
+  }
+
+  private _handelCreatedChannels() {
```
`handle`

shawnm0705 commented on src/app/chat/chat-list/chat-list.component.ts:
```diff
@@ -172,50 +175,28 @@ export class ChatListComponent {
-        this._groupingChatChannels();
+    await modal.present();
+    modal.onWillDismiss().then((data) => {
+      if (data.data && data.data.newChannel) {
+        if (!this._channelExist(data.data.newChannel, 'cohort')) {
+          this.chatChannels.push(data.data.newChannel);
+          this._groupingChatChannels();
+          // Subscribe to the pusher channel of new create chat channel.
+          this._checkAndSubscribePusherChannels();
+          this.goToChatRoom(data.data.newChannel);
```
Overall, don't do this kind of modifications after we create a new channel. It's so easy to make mistake.
Instead, we can just retrieve the chat list again from API.

sasangachathumal committed:
[CUT-132] fixed typo in method name.

shawnm0705 commented on src/app/chat/chat-list/chat-list.component.ts:
```diff
@@ -277,4 +264,65 @@ export class ChatListComponent {
+    });
+    await modal.present();
+    modal.onWillDismiss().then((data) => {
+      if (data.data && (data.data.learnerChannel || data.data.expertChannel)) {
```
We don't need `&& (data.data.learnerChannel || data.data.expertChannel)`

sasangachathumal committed:
[CUT-132] remove extra logic from condition.
